### ISSUE TITLE
It seems GUFA pass use constant propagation for exported mutable globals which is wrong

### ISSUE THREAD

**Issue by MaxGraey** [WebAssembly/binaryen#4932]
original:
```wat
(module
 (global $foo (mut i32) (i32.const 0))  ;; exported mutable global
 (export "foo" (global $foo))
 (export "test" (func $test))
 (func $test (result i32)
  global.get $foo
 )
)
```
After GUFA pass (which contain issue):
```wat
(module
 (global $foo (mut i32) (i32.const 0))
 (export "foo" (global $foo))
 (export "test" (func $test))
 (func $test (result i32)
  i32.const 0 ;; this is invalid due to $foo can be changed externally
 )
)
```
expected:
```wat
(module
 (global $foo (mut i32) (i32.const 0))
 (export "foo" (global $foo))
 (export "test" (func $test))
 (func $test (result i32)
  global.get $foo ;; preserve read from exported mutable global
 )
)
```

**Comment by kripken** [comment]
Thanks! Indeed this is a bug. Fix in #4935

**Comment by MaxGraey** [comment]
Thanks!
